osquery/events/linux/http_event_publisher.cpp

```diff
@@ -0,0 +1,863 @@
+/**
+ *  Copyright (c) 2014-present, Facebook, Inc.
+ *  All rights reserved.
+ *
+ *  This source code is licensed under both the Apache 2.0 license (found in the
+ *  LICENSE file in the root directory of this source tree) and the GPLv2 (found
+ *  in the COPYING file in the root directory of this source tree).
+ *  You may select, at your option, one of the above-listed licenses.
+ */
+
+#include "http_event_publisher.h"
+#include "httpparser/httprequestparser.h"
+#include "httpparser/httpresponseparser.h"
+#include "httpparser/request.h"
+#include "httpparser/response.h"
+#include "osquery/remote/serializers/json.h"
+#include <arpa/inet.h>
+#include <boost/range/adaptor/map.hpp>
+#include <netinet/if_ether.h>
+#include <netinet/in.h>
+#include <netinet/ip.h>
+#include <netinet/ip6.h>
+#include <osquery/utils/conversions/split.h>
+#include <osquery/utils/conversions/tryto.h>
+#include <osquery/flags.h>
+#include <osquery/hashing/hashing.h>
+#include <osquery/logger.h>
+#include <osquery/registry_factory.h>
+#include <set>
+#include <sys/socket.h>
+#include <vector>
+
+namespace osquery {
+#define IPV6_VERSION		0x60   
+#define IPV4 IPVERSION
+#define IPV6 (IPV6_VERSION >> 4)
+
+typedef std::pair<std::string, std::string> LocalRemoteAddrs;
+    
+static const size_t kIPv6Length = sizeof(struct ip6_hdr);
+    
+/// Maximum bytes per packet.
+static const int kSnapLength = 1518;
+
+/// Avoid running pcap in a busy loop.
+static const int kPacketBufferTimeoutMs = 1000;
+
+/// Internal traffic filter
+const std::string kInternalTrafficFilter =
```

Comment:
Why? I can easily imagine wanting this information for nominally internal space. If you think this is valuable, what is the ipv6 blacklist?

Reply:
yeah, I felt it is important to monitor internal traffic as well, so I removed the code which filters out the internal traffic.